File: ColaBottle.ts
class ColaBottle implements Bottle {
    private soda: Soda = null;
    private label: Label = null;
    private closer: Closer = null;

    constructor() {
        console.log("Создали бутылку Cola");
    }

    //Override
    public pourSoda(soda: Soda): void {
        if (this.closer == null) {
            this.soda = soda;
            console.log(`Налили газировку ${this.soda}`);
        } else {
            console.log("Сорри... Бутылка закрыта");
        }
    }

    //Override
    public putLabel(label: Label): void {
        this.label = label;
        console.log(`Наклеили этикетку ${this.label}`);
    }

    //Override
    public seal(closer: Closer): void {
        this.closer = closer;
        console.log(`Закрыли крышкой ${this.closer}`);
    }

    //Override
    public copy(): Bottle {
        let result: ColaBottle = new ColaBottle();
        result.soda = this.soda.copy();
        result.label = this.label.copy();
        result.closer = this.closer.copy();
        return result;
    }
}

File: тз/ColaBottle.java
package ru.agentd;

public class ColaBottle implements Bottle {
    private Soda soda = null;
    private Label label = null;
    private Closer closer = null;

    public ColaBottle() {
        System.out.println("Создали бутылку Cola");
    }

    @Override
    public void pourSoda(Soda soda) {
        if (closer == null) {
            this.soda = soda;
            System.out.println("Налили газировку");
        } else {
            System.out.println("Бутылка закрыта");
        }
    }

    @Override
    public void putLabel(Label label) {
        this.label = label;
        System.out.println("Наклеили этикетку");
    }

    @Override
    public void seal(Closer closer) {
        this.closer = closer;
        System.out.println("Закрыли крышкой");
    }

    @Override
    public Bottle copy() {
        ColaBottle result = new ColaBottle();
        result.soda = soda.copy();
        result.label = label.copy();
        result.closer = closer.copy();
        return result;
    }
}
